js/containers/report/iframe-standalone-app.js

```diff
@@ -0,0 +1,104 @@
+  }
+
+  renderIframe() {
+    const { report, iframeQuestionId, answers } = this.props;
+
+    const question = report.get("questions").get(iframeQuestionId);
+
+    const studentId = config("studentId");
+    const answer = answers.filter(a =>
+      a.get("questionId") === iframeQuestionId &&
+      a.get("platformUserId") === studentId
+    ).first();
+
+    if (!answer) {
+      const errorText =
+        !iframeQuestionId ? "Parameter 'iframeQuestionId' is missing" :
+        !studentId ? "Parameter 'studentId' is missing" :
+        `No data for question '${iframeQuestionId}' by student '${studentId}'`;
+      return <DataFetchError error={{title: "Unable to fetch data", body: errorText}} />;
+    }
+
+    let url;
+    let state;
+    // There are two supported answer types handled by iframe question: simple link or interactive state.
+    if (answer.get("type") === "external_link") {
+      // Answer field is just the reportable URL. We don't need any state.
+      url = answer.get("answer");
+      state = null;
+    } else if (answer.get("type") === "interactive_state") {
+      // URL field is provided by question. Answer field is a state that will be passed
+      // to the iframe using iframe-phone.
+      url = question.get("url");
+      state = answer.get("answer");
+    }
+    return (
+      <InteractiveIframe src={url} state={state} style={{border: "none"}} width="100%" height="100%" />
+    );
+  }
+
+  render() {
+    const { report, error, isFetching } = this.props;
+    return (
+      <div className="report-app full-size">
+        <div className="report full-size" style={{ opacity: isFetching ? 0.3 : 1 }}>
+          {report && this.renderIframe()}
+          {error && <DataFetchError error={error} />}
+        </div>
+        {isFetching && <LoadingIcon />}
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  const data = state.get("data");
+  const error = data.get("error");
+  const reportState = state.get("report");
+  const iframeQuestionId = reportState && reportState.get("iframeQuestionId");
+  const answers = reportState && reportState.get("answers");
+  const dataDownloaded = !error && !data.get("isFetching");
+  return {
+    report: dataDownloaded && reportState,
+    iframeQuestionId: iframeQuestionId,
+    answers,
+    isFetching: data.get("isFetching"),
+    error,
+  };
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    fetchAndObserveData: () => dispatch(fetchAndObserveData()),
+    hideUnselectedQuestions: () => dispatch(hideUnselectedQuestions()),
+    showUnselectedQuestions: () => dispatch(showUnselectedQuestions()),
+    setNowShowing: (nowShowingValue, selectedStudentIds) => dispatch(setNowShowing(nowShowingValue, selectedStudentIds)),
+    setAnonymous: value => dispatch(setAnonymous(value)),
+    hideCompareView: () => dispatch(hideCompareView()),
+    trackEvent: (category, action, label) => dispatch(trackEvent(category, action, label)),
+  };
+};
```

Comment:
Not sure how easy this is to refactor but it looks identical to: https://github.com/concord-consortium/portal-report/blob/172324236-standalone-iframe/js/containers/report/report-app.js#L95